x/compliance/keeper/msg_server_provision_model.go

```diff
@@ -69,6 +69,13 @@ func (k msgServer) ProvisionModel(goCtx context.Context, msg *types.MsgProvision
 	// store compliance info
 	k.SetComplianceInfo(ctx, complianceInfo)
 
+	deviceSoftwareCompliance, _ := k.GetDeviceSoftwareCompliance(ctx, msg.CDCertificationId)
```

Comment:
I think we should not update deviceSoftwareCompliance list for the provisioned state. Only for the certified state. 
So, this code can be removed.

Reply:
Done! ✅